python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -360,6 +377,110 @@ cdef class Column:
             []
         )
 
+    @classmethod
+    def from_ndarray(cls, obj):
+        """
+        Create a Column from any object which supports the NumPy array interface.
+
+        Parameters
+        ----------
+        obj : Any
+            The input array to be converted into a `pylibcudf.Column`.
+
+        Returns
+        -------
+        Column
+
+        Raises
+        ------
+        TypeError
+            If the input type is neither `numpy.ndarray` nor `cupy.ndarray`.
+        ImportError
+            If NumPy or CuPy is required but not installed.
+
+        Notes
+        -----
+        If `obj` is a 2D CuPy array, the resulting column is a list column.
+        NumPy conversion logic is not yet implemented.
+        Multi-dimensional arrays (ndim > 2) are not supported.
+        """
+        return cls._from_ndarray(obj)
+
+    @singledispatchmethod
+    @classmethod
+    def _from_ndarray(cls, obj):
+        if np_error is not None:
+            raise np_error
+        if cp_error is not None:
+            raise cp_error
+        raise TypeError(f"Cannot convert a {type(obj)} to a pylibcudf Column")
```

Comment:
This raises too eagerly in the case where we're trying to convert from a numpy array and the user has numpy but not cupy.